
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Table of Contents &#8212; peddy 0.4.3 documentation</title>
    <link rel="stylesheet" href="_static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="CSV Output" href="output.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="#">
          <span>peddy 0.4.3 documentation</span></a></h1>
        <h2 class="heading"><span>Table of Contents</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        <a class="uplink" href="#">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="output.html">CSV Output</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <div class="section" id="table-of-contents">
<h1>Table of Contents<a class="headerlink" href="#table-of-contents" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="output.html">CSV Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="html.html">Example HTML</a></li>
<li class="toctree-l1"><a class="reference internal" href="resolve.html">Error Resolution</a></li>
</ul>
</div>
</div>
<div class="section" id="manipulation-validation-and-exploration-of-pedigrees">
<h1>manipulation, validation and exploration of pedigrees<a class="headerlink" href="#manipulation-validation-and-exploration-of-pedigrees" title="Permalink to this headline">¶</a></h1>
<p>peddy compares familial-relationships and sexes as reported in a <a class="reference external" href="https://www.cog-genomics.org/plink2/formats#fam">PED/FAM file</a>
with those inferred from a VCF.</p>
<p>It samples the VCF at about 25000 sites (plus chrX) to accurately estimate <strong>relatedness</strong>, <strong>IBS0</strong>, <strong>heterozygosity</strong>, <strong>sex</strong> and <strong>ancestry</strong>. It uses 2504 thousand genome samples as backgrounds to calibrate the relatedness calculation and to make ancestry predictions.</p>
<p>It does this very quickly by sampling, by using C for computationally intensive parts, and by parallelization.</p>
<p>The command-line usage looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python -m peddy -p <span class="m">4</span> --plot ceph1463.vcf.gz ceph1463.ped
</pre></div>
</div>
<p>This will use 4 cpus to run various checks and create <a class="reference external" href="_static/ceph.html">ceph1463.html</a> which
you can open in any browser to interactively explore your data. Unless you have triple digit numbers
of samples, using more than 4 cpus will give only marginal improvement.</p>
<p>It will also create create 4 csv files and 4 static QC plots that mirror those in the interactive html.
These will indicate:</p>
<ul class="simple">
<li><p>discrepancies between ped-reported and genotype-inferred relations</p></li>
<li><p>discrepancies between ped-reported and genotype-inferred sex</p></li>
<li><p>samples with higher levels of HET calls, lower depth, or more variance in b-allele-frequency (alt / (ref + alt )) for het calls.</p></li>
<li><p>an ancestry prediction based on projection onto the thousand genomes principal components</p></li>
</ul>
<p>Finally, it will create a new file ped file <cite>ceph1463.peddy.ped</cite> that also lists
the most useful columns from the <cite>het-check</cite> and <cite>sex-check</cite>. Users can <strong>first
look at this extended ped file for an overview of likely problems</strong>.</p>
<p>The columns in the CSV output are documented in <a class="reference internal" href="output.html#output"><span class="std std-ref">CSV Output</span></a></p>
<div class="section" id="static-images">
<h2>Static Images<a class="headerlink" href="#static-images" title="Permalink to this headline">¶</a></h2>
<p>This will create a number of images:</p>
<div class="section" id="sex-check">
<h3>Sex Check<a class="headerlink" href="#sex-check" title="Permalink to this headline">¶</a></h3>
<p>A sex-check assumes that males should have very few heterozygote calls on the X-
chromosome and females should have relatively many. Here, we see, as expected
that there are no sex issues in the CEPH cohort:</p>
<img alt="_images/ceph.sex_check.png" src="_images/ceph.sex_check.png" />
<p>If there are samples with unspecified sex in the ped file, they will appear
in the center of the plot as ‘unknown’.</p>
</div>
<div class="section" id="het-check">
<h3>Het Check<a class="headerlink" href="#het-check" title="Permalink to this headline">¶</a></h3>
<p>The het check looks for samples with higher rates of het calls, usually, this
can indicate sample contamination. This plot also shows depth along the X-axis
as a way to quickly check for samples with lower coverage.</p>
<img alt="_images/ceph.het_check.png" src="_images/ceph.het_check.png" />
</div>
<div class="section" id="ancestry-check">
<h3>Ancestry Check<a class="headerlink" href="#ancestry-check" title="Permalink to this headline">¶</a></h3>
<p>Since we know the ancestry of the thousand genomes samples
we can project the current peddy input (in this case CEPH)
onto the principal components of the thousand genomes samples
and then predict the ancestry of incoming samples:</p>
<img alt="_images/ceph.pca_check.png" src="_images/ceph.pca_check.png" />
<p>Note that, as expected all of the members of the CEPH pedigree
are of ‘EUR’opean descent.</p>
</div>
<div class="section" id="relatedness-check">
<h3>Relatedness Check<a class="headerlink" href="#relatedness-check" title="Permalink to this headline">¶</a></h3>
<p>In this check, we compare the relatedness specified in the pedigree
file to what is observed by the genotypes. For example, a sib-sib
pair should have a relatedness coefficient of 0.5. In the plot,
<strong>sample-pairs are *colored* according to their expected relatedness
specified in the ped file and *located* in the plot according to their
relatedness (and IBS levels) calculated from the genotypes</strong></p>
<p>IBS0 is the number of sites for which the 2 samples shared 0 alleles.
For parent-child pairs and IBS0 event is a (putative) <em>de novo</em> and so should
happen very infrequently. Unrelated samples should have a relatedness of 0
and a higher IBS0.</p>
<p>IBS2 is the number of sites where the 2 samples are both het or
both homozygous alternate.</p>
<img alt="_images/ceph.ped_check.png" src="_images/ceph.ped_check.png" />
</div>
<div class="section" id="csvs">
<h3>CSVs<a class="headerlink" href="#csvs" title="Permalink to this headline">¶</a></h3>
<p>For each of those images, there is a corresponding <cite>.csv</cite> file.
See <a class="reference internal" href="output.html#output"><span class="std std-ref">CSV Output</span></a> for a description of the columns.</p>
</div>
</div>
</div>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        <a class="uplink" href="#">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="output.html">CSV Output</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016, Brent S Pedersen.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.1.2.
    </div>
  </body>
</html>